src/lib/regex.js

```diff
@@ -0,0 +1,23 @@
+/**
+* Copyright 2012-2017, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+// Simple helper functions
+// none of these need any external deps
+
+/*
+ * make a regex for matching counter ids/names ie xaxis, xaxis2, xaxis10...
+ *  eg: regexCounter('x')
+ * tail is an optional piece after the id
+ *  eg regexCounter('scene', '.annotations') for scene2.annotations etc.
+ */
+exports.counter = function(head, tail, openEnded) {
+    return new RegExp('^' + head + '([2-9]|[1-9][0-9]+)?' +
```

Comment:
Ah, so `xaxis0` and `xaxis1` are strictly invalid? 0 makes sense. 1 seems slightly asymmetric, but nothing new.

Reply:
We sanitize a few of these (turning `scene1` into `scene` etc) on the way in using [`cleanLayout`](https://github.com/plotly/plotly.js/blob/11bb997e78128b9f47db2ed392982f82a026e589/src/plot_api/helpers.js#L56) but yeah, it's always been that way. The idea is that most of the time you don't have more than one, so needing to add the `1` would be annoying and ugly.

Reply:
We should perhaps try to sanatize all subplots attribute names in `cleanLayout`. Looks like we only sanatize cartesian, gl2d and gl3d axes at the moment.